addons/Hierarchical_Lesson_Report/src/presenter.js

```diff
@@ -72,135 +123,308 @@ function AddonHierarchical_Lesson_Report_create() {
+        var calculateScore = !(isPreview || (isInPrintableMode && isInPrintableEmptyState));
+
+        createProgressCell($row, {
+            score: calculateScore ? presenter.calculateLessonScaledScore() : 0,
+            count: 1
+        }, undefined, undefined, isInPrintableMode);
+    }
+
+    function isPreviewConsideringPrintableState(isInPrintableMode) {
+        return isInPrintableMode
+           ? printableController.isPreview()
+           : presenter.isPreview;
+    }
+
+    presenter.calculateLessonScaledScore = function () {
+        var lessonScore = getLessonScore();
+        if (lessonScore.pageCount === 0) {
+            return 0;
         }
+
+        return Math.floor((lessonScore.scaledScore / lessonScore.pageCount) * 100) / 100;
+    };
+
+    function addPageScoreFooterCell($cell, lessonScore, isInPrintableMode) {
+        var $separator = generateSeparator(isInPrintableMode);
+        var content = lessonScore.score + $separator[0].outerHTML + lessonScore.maxScore;
+        addPageScoreCell($cell, content, isInPrintableMode);
     }
 
-    function createRow(index, parentIndex, isChapter) {
+    function createRow($view, index, parentIndex, isChapter, configuration, isInPrintableMode) {
         var row = document.createElement('tr');
+        $(row).appendTo($view.find('table'));
+        $(row).addClass(configuration.classes[index % configuration.classes.length]);
 
-        $(row).appendTo($("#" + presenter.treeID).find('table'));
-        $(row).addClass("treegrid-" + index);
-        $(row).addClass(presenter.configuration.classes[index % presenter.configuration.classes.length]);
+        var nodeCssClass = isInPrintableMode
+          ? CSS_CLASSES.PRINTABLE_HIER_REPORT_NODE + '-' + index
+          : "treegrid-" + index;
+        $(row).addClass(nodeCssClass);
 
         if (parentIndex != null) {
-            $(row).addClass("treegrid-parent-" + parentIndex);
+            var parentNodeCssClass = isInPrintableMode
+              ? CSS_CLASSES.PRINTABLE_HIER_REPORT_PARENT + '-' + parentIndex
+              : "treegrid-parent-" + parentIndex
+            $(row).addClass(parentNodeCssClass);
         }
 
-        if (isChapter) {
-            $(row).addClass("hier_report-chapter");
-        } else {
-            $(row).addClass(index % 2 > 0 ? "hier_report-odd" : "hier_report-even");
-        }
+        var isOdd = index % 2 > 0;
+        var typeCssClass = isChapter
+          ? getRowChapterCSSCLass(isInPrintableMode)
+          : getRowPageCSSClass(isOdd, isInPrintableMode);
+        $(row).addClass(typeCssClass);
 
         return row;
     }
 
-    function createProgressCell(row, score, index, isChapter) {
+    function getRowChapterCSSCLass(isInPrintableMode) {
+        return isInPrintableMode
+            ? CSS_CLASSES.PRINTABLE_HIER_REPORT_CHAPTER
+            : CSS_CLASSES.HIER_REPORT_CHAPTER;
+    }
+
+    function  getRowPageCSSClass(isOdd, isInPrintableMode) {
+        if (isOdd) {
```

Comment:
Alternatywna propozycja, która pozwala się pozbyć zagnieżdżeń i moim zdaniem uprościć wizję. Do poprawy według uznania dewelopera ;)
`var class = isInPrintableMode ? "PRINTABLE_" : "";
class += "HIER_REPORT_";
class += isOdd ? "ODD" : "EVEN";
return CSS_CLASSES[class]; `

Reply:
Takiego rozwiązania staram się unikać. Uważam, że o ile nie będzie to konieczne to lepiej posługiwać się zmiennymi a nie stringami. Ułatwia to wyszukiwanie w pliku zarówno przed użytkownika i IDE.